spec/app_support/order_row_importer_spec.rb

```diff
@@ -0,0 +1,271 @@
+
+      it "flags the fulfillment date as invalid" do
+        subject.import
+        expect(subject.errors)
+          .to include_a_string_matching(/Invalid Fulfillment Date/)
+      end
+    end
+
+    context "when the order date is invalid" do
+      it_behaves_like "an order was not created"
+
+      it "flags the order date as invalid" do
+        subject.import
+        expect(subject.errors)
+          .to include_a_string_matching(/Invalid Order Date/)
+      end
+    end
+
+    context "when the user is invalid" do
+      context "when looking up by username" do
+        it_behaves_like "an order was not created"
+
+        it "flags the username as invalid" do
+          subject.import
+          expect(subject.errors)
+            .to include_a_string_matching(/Invalid username/)
+        end
+      end
+
+      context "when looking up by email address" do
+        it_behaves_like "an order was not created"
+
+        it "flags the username as invalid" do
+          subject.import
+          expect(subject.errors)
+            .to include_a_string_matching(/Invalid username/)
+        end
+      end
+    end
+
+    context "when the product name is invalid" do
+      it_behaves_like "an order was not created"
+
+      it "flags the product name as invalid" do
+        subject.import
+        expect(subject.errors)
+          .to include_a_string_matching(/Couldn't find product by name/)
+      end
+    end
+
+    context "when the product is a service" do
+      let(:chart_string) { account.account_number }
+      let(:price_group) { create(:price_group, facility: facility) }
+      let(:product) { service }
+      let(:product_name) { product.name }
+      let(:username) { user.username }
+
+      before(:each) do
+        create(:user_price_group_member, user: user, price_group: price_group)
+        product.service_price_policies.create(
+          attributes_for(:service_price_policy, price_group: price_group)
+        )
+      end
+
+      context "and it requires a survey" do
+        before { Service.any_instance.stub(:active_survey?).and_return(true) }
+
+        it_behaves_like "an order was not created"
+
+        it "flags the service as requiring a survey" do
+          subject.import
+          expect(subject.errors)
+            .to include_a_string_matching(/Service requires survey/)
+        end
+      end
+
+      context "and it requires a template" do
+        before { Service.any_instance.stub(:active_template?).and_return(true) }
+
+        it_behaves_like "an order was not created"
+
+        it "flags the service as requiring a template" do
+          subject.import
+          expect(subject.errors)
+            .to include_a_string_matching(/Service requires template/)
+        end
+      end
+    end
+
+    context "when the user has an account for the product's facility" do
+      let(:chart_string) { account.account_number }
+      let(:fulfillment_date) { "1/1/1999" }
```

Comment:
Not sure if we have some other tests around it, but we should validate that something like `1/10/1999` is January 1 and not October 1. Also, we might want some specs around `1/1/99`. Can't remember if that's supposed to be an error or not.